mesonbuild/backend/ninjabackend.py

```diff
@@ -237,8 +237,10 @@ def _quoter(x, qf = quote_func):
         return ninja_quote(qf(str(x)))
 
     def write(self, outfile: T.TextIO) -> None:
+        rspfile_args = self.args
         if self.rspfile_quote_style is RSPFileSyntax.MSVC:
             rspfile_quote_func = cmd_quote
+            rspfile_args = [NinjaCommandArg('$in_newline', arg.quoting) if arg.s == '$in' else arg for arg in rspfile_args]
```

Comment:
Why not just replace it where the `$in` is generated?

Reply:
Because as far as I can tell the RSP rule is a duplicate of the regular rule, the generator that adds `$in` is in a state unaware of whether or not RSP is used. 

There would be a less than insignificant refactor to the rule writer because the lines https://github.com/mesonbuild/meson/blob/523a27c6f0b40c7ed1cc04d4d5c2d74cc9f6b30e/mesonbuild/backend/ninjabackend.py#L245-L251 assume the command is identical for both RSP and regular rules. 

In other words, everything related to the RSP generation would have to be duplicated into every generator that calls `self._rsp_options(compiler)` right before calling `self.add_rule(NinjaRule(...` and the writer would have to be refactored.